uaclient/clouds/identity.py

```diff
@@ -0,0 +1,55 @@
+import json
+
+from uaclient import exceptions
+from uaclient import clouds
+from uaclient import status
+from uaclient import util
+
+CLOUDINIT_RESULT_FILE = "/var/lib/cloud/data/result.json"
+
+
+# Mapping of datasource names to cloud-id responses. Trusty compat with Xenial+
+DATASOURCE_TO_CLOUD_ID = {"ec2": "aws"}
+
+
+def get_cloud_type_from_result_file(result_file=CLOUDINIT_RESULT_FILE) -> str:
+    result = json.loads((util.load_file(result_file)))
+    dsname = result["v1"]["datasource"].split()[0].lower()
```

Comment:
How will this behave if cloud-init fails for some reason? I _assume_ that these keys would only be missing if something really bad happened (which would presumably mean that a premium attach either won't work or will be pointless because the machine is going to have to be decommissioned)?

Reply:
There would be a couple reasons for result.json not being the format we expect or not being present:
 - Cloud-init version is much older than what is shipped in trusty-updates
 - cloud-init failed  on a system due to missing dependencies
 - cloud-init failed on an unsupported platform or distro type

Any of these would fall into the unsupported images camp, so I don't think we need to address that.